shared/src/main/scala/coop/rchain/shared/ScalaCheckOps.scala

```diff
@@ -0,0 +1,22 @@
+package coop.rchain.shared
+
+import cats.syntax.all._
+import monix.eval.Task
+import org.scalacheck.effect.PropF
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.{Assertion, Assertions}
+
+/**
+  * Wrappers for writing property-based tests using the ScalaCheck Effect library.
+  */
+object ScalaCheckOps {
+
+  def forAllF[T: Arbitrary, P](f: T => Task[P]): Task[Assertion] =
+    PropF.forAllF(f.map(_.void)).check().map(r => Assertions.assert(r.passed, r.status.toString))
+
+  def forAllF[T: Arbitrary, P](data: Gen[T])(f: T => Task[P]): Task[Assertion] =
+    PropF
+      .forAllF(data)(f.map(_.void))
+      .check()
+      .map(r => Assertions.assert(r.passed, r.status.toString))
```

Comment:
This is good starting point for the solution but it only works for one Arbitrary argument so to make work in more cases we need to replicate the similar code definitions as in Prop.forAll or PropF.forAll where is defined for up to 8 arguments.
https://github.com/typelevel/scalacheck-effect/blob/f146a3422d0/core/shared/src/main/scala/org/scalacheck/effect/PropF.scala#L597-L931

Reply:
It seems that `check` from Checkers is defined in a similar way which means you are on the good track for the solution.
```scala
def check[A1, P](f: A1 => P, configParams: PropertyCheckConfigParam*)
  (implicit
    config: PropertyCheckConfiguration,
    p: P => Prop,
    a1: Arbitrary[A1], s1: Shrink[A1], pp1: A1 => Pretty,
    prettifier: Prettifier,
    pos: source.Position
  ): Assertion = {
  val params = getScalaCheckParams(configParams, config)
  asserting.check(Prop.forAll(f)(p, a1, s1, pp1), params, prettifier, pos)
}
```
https://github.com/scalatest/scalatestplus-scalacheck/blob/32e7dc67a/scalatestPlusScalaCheck/src/main/scala/org/scalatestplus/scalacheck/Checkers.scala#L235-L245
We can replicate these definitions for PropF.forAll. It would be also a good addition to scalacheck-effect library as well.